plugin/src/ast_helpers/number.rs

```diff
@@ -5,37 +5,81 @@ use tree_sitter_lint::{tree_sitter::Node, QueryMatchContext};
 
 use crate::{assert_kind, kind};
 
+#[derive(Copy, Clone, Debug)]
+pub enum NumberOrBigInt {
```

Comment:
So it turns out that BigInt's can't eg be compared to numbers. So it seemed to make sense (given that eg the tree-sitter Javascript grammar uses `number` for both numbers and big-int literals) to have things like parsing the numeric value of a node now target a `NumberOrBigInt` enum instead

And so then sort of selectively modified/moved stuff from `Number` -> `NumberOrBigInt` as needed